File: producer_java/src/test/java/com/example/BeerStuffBase.java
package com.example;

//remove::start[]

import io.restassured.config.EncoderConfig;
import io.restassured.module.mockmvc.RestAssuredMockMvc;
import io.restassured.module.mockmvc.config.RestAssuredMockMvcConfig;
import org.junit.jupiter.api.BeforeEach;
//remove::end[]

public abstract class BeerStuffBase {
	//remove::start[]

	@BeforeEach
	public void setup() {
		// https://github.com/spring-cloud/spring-cloud-contract/issues/1428
		EncoderConfig encoderConfig = new EncoderConfig().appendDefaultContentCharsetToContentTypeIfUndefined(false);
		RestAssuredMockMvc.config = new RestAssuredMockMvcConfig().encoderConfig(encoderConfig);
		RestAssuredMockMvc.standaloneSetup(new StuffController());
	}
	//remove::end[]
}

File: producer_kotlin/src/test/kotlin/com/example/BeerStuffBase.kt
package com.example

import io.restassured.config.EncoderConfig
import io.restassured.module.mockmvc.RestAssuredMockMvc
import io.restassured.module.mockmvc.config.RestAssuredMockMvcConfig
import org.junit.jupiter.api.BeforeEach

abstract class BeerStuffBase {
    @BeforeEach
    fun setup() {
		val encoderConfig: EncoderConfig = EncoderConfig().appendDefaultContentCharsetToContentTypeIfUndefined(false)
		val restAssuredConf: RestAssuredMockMvcConfig = RestAssuredMockMvcConfig().encoderConfig(encoderConfig)
		RestAssuredMockMvc.config = restAssuredConf
		RestAssuredMockMvc.standaloneSetup(StuffController())
    }
}
